trie.c

```diff
@@ -75,17 +75,38 @@ void st_get_payload_from_code(st_trie_payload_t *payload, uint8_t code_byte1, ui
 {
     // Payload data is bit-backed into 16bits:
     // (N: node type, F: func, B: backspaces, C: completion length)
-    // 0b NNFF BBBB CCCC CCCC
-    payload->func_code = (code_byte1 >> 4) & 3;
-    payload->num_backspaces = code_byte1 & 15;
+    // 0b NFFB BBBB CCCC CCCC
+    payload->func_code = (code_byte1 >> 5) & 0x03;
+    payload->num_backspaces = code_byte1 & 0x1f;
     payload->completion_len = code_byte2;
     payload->completion_index = completion_index;
 }
+//////////////////////////////////////////////////////////////////
+void st_get_node_info(const st_trie_t *trie, st_trie_node_info_t *node_info, uint16_t *offset)
```

Comment:
The node header was formerly just the first two bits of the first byte in a node, and we manually parsed them right in the trie traversal code. With the addition of chained rule matches, we need more than two bits of information to know how to parse the rest of the node.

The first two bits serve basically the same purpose as before.

- Bit 7 of the first byte represents that there is some kind of match on this node (could be a regular match, one or more chained matches, or both).
- Bit 6 represents that there is a branch or (if bit 7 is also set) that there are longer rules to continue searching for.
- Bit 5 is on if there is an unchained match (this is the original kind of match)
- Bit 4 signals if the chain check count is contained in the next 4 bits, or if there are too many, and it is instead represented by the next 12 bits (adding a second byte to the header).
  - If bit 4 is off, then bits 3..0 represent the chain check count as a number from 0 to 15.
  - If bit 4 is on, then bits 3..0 and the entire next byte represent the chain check count as a number from 0 .. 4095.

This function parses all of that into a struct and advances the trie offset to the first byte after the header, so the rest of the code doesn't need to care how long the header was.
- Bit 3..0